slice-core/src/main/java/com/cognifide/slice/core/internal/module/SliceResourceModule.java

```diff
@@ -88,12 +96,44 @@ public void afterInjection(Object injectee) {
 			if (resource == null) {
 				return;
 			}
+
+			invokeMethod(injectee, PreMapping.class);
 			Mapper mapper = mapperProvider.get();
+			invokeMethod(injectee, PostMapping.class);
+
 			mapper.get(resource, injectee);
 			if (injectee instanceof InitializableModel) {
 				((InitializableModel) injectee).afterCreated();
 			}
 		}
+
+		private void invokeMethod(final Object injectee, final Class<? extends Annotation> annotationClass) {
+			Method method = findMethod(injectee, annotationClass);
```

Comment:
It will be very inefficient to look for annotated method every time an object is created. We need to revisit this solution and use some kind of caching.